### ISSUE TITLE
multiple connections

### ISSUE THREAD

**Issue by Awesomestorm** [getnamo/socketio-client-ue4#38]
I updated the plugin to latest version 0.5.2 and that appears to have sorted out the ue4 editor freezing everytime I was coming out of the play editor. There is however another issue that has come up, whenever I launch the server, after my : listening on port message I am getting multiple client connections - up to 20 even though I have not lauched the play editor in ue4.

**Comment by getnamo** [comment]
Will need a bit more detailed run through of how to replicate the issue you're seeing. Walk through an example step by step

**Comment by Awesomestorm** [comment]
For testing purposes I simply created a new third person template, and
placed an actor in the level with this script:
​and setup a very simple script inside nodejs:

​I don't know if this may be a factor but I use gitbash to launch the
server with the nodemon command and this is the result without doing
anything inside of unreal (play in editor).

​

**Comment by Awesomestorm** [comment]
Sorry I should add that the disconnects in the command prompt are a result
of playing in the editor and then stopping play

**Comment by Awesomestorm** [comment]
Update. I appear to have sorted the issue by firstly removing the test actor from the level and then in my character BP I disabled the socket io auto connect and then off of the event begin play I did an authority check and off of the socket ioclient I called connect, added the http to it and that seems to have worked.

**Comment by getnamo** [comment]
the plugin does connect once per SocketIOComponent, this is expected behavior, question is if it's connecting more than once per component and if so, what is the minimal reproducible example?

**Comment by Awesomestorm** [comment]
the only min reproducible example is the one that I gave you, I have not tried any others in engine version 4.15.1 and the solution that appears to be working (posted in the comments above) where there is no connections on starting the server unless I launch the play editor, then it gives the right amount of connects. If I enable auto connect it appears to be a 50/50 situation, sometimes it multiple connects and others not :( perhaps it is not the plugin but an engine bug.

**Comment by getnamo** [comment]
Note in the issues for this github https://github.com/getnamo/socketio-client-ue4/issues/38 I don't see anything beyond this for your description: 

_For testing purposes I simply created a new third person template, and
placed an actor in the level with this script:
​and setup a very simple script inside nodejs:_

_​I don't know if this may be a factor but I use gitbash to launch the
server with the nodemon command and this is the result without doing
anything inside of unreal (play in editor)._

Don't have anything underneath _this script:_ so it's hard for me to picture what you're doing, need ideally exact steps to replicate or a minimum zipped up project so I can replicate it.

**Comment by Awesomestorm** [comment]
ok let me try again...
For testing purposes I simply created a new third person template, and
placed an actor in the level with this script:
![actorscript](https://cloud.githubusercontent.com/assets/17770345/24827643/93f6a6a2-1c45-11e7-88ad-07a961d715b6.jpg)

and setup a very simple script inside nodejs:

![nodescript](https://cloud.githubusercontent.com/assets/17770345/24827651/aece71da-1c45-11e7-9525-3fbf72389757.jpg)
I don't know if this may be a factor but I use gitbash to launch the
server with the nodemon command and this is the result without doing
anything inside of unreal (play in editor).
![serverlaunchresult](https://cloud.githubusercontent.com/assets/17770345/24827659/d17338ba-1c45-11e7-9093-32a3e5a33bcf.jpg)
The disconnects in the command prompt are a result of playing in the editor and then stopping play ( ue4 editor play settings at this point on 2 and dedicated server).

**Comment by getnamo** [comment]
candidate fix merged in: https://github.com/getnamo/socketio-client-ue4/commit/d7b3d81201c685d4f5e01e53373ecebcc3d7f83b, test the new release and see if it has been fixed for you: https://github.com/getnamo/socketio-client-ue4/releases/tag/0.6.0
